### PR TITLE
Add debug-info to access variables from generator state

### PR DESCRIPTION
**Pull request by Nemo157**

### CODE REVIEW

**Comment by Nemo157** [comment]
I previously ran the tests when I first implemented this, but since then I've got a new laptop and haven't been able to get `debuginfo` tests running properly; hopefully Travis is happy (although, I just figured out a way to manually test it).

This will make all the variables in the generator available even when they haven't been initialized yet (or after they've been de-initialized), I'm not sure if there's an easy way to make it so the correct variables are specified in each state.

File: src/librustc_codegen_ssa/mir/mod.rs [hunk 1]
```diff
@@ -636,6 +636,44 @@ fn arg_local_refs<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>>(
...
+                        &fx.debug_context,
+                        name,
+                        ty,
+                        scope,
+                        variable_access,
+                        VariableKind::LocalVariable,
+                        DUMMY_SP
+                    );
+                }
+            }
```
Comment (Zoxc) [inline src/librustc_codegen_ssa/mir/mod.rs]: Probably should factor this out to a method instead of duplicating it

**Comment by Centril** [comment]
Ping from triage @Nemo157 :)

File: src/librustc_codegen_ssa/mir/mod.rs (hunk 1 above)
Comment (Nemo157) [reply src/librustc_codegen_ssa/mir/mod.rs]: Updated to first combine the `UpvarDecl`s and `LocalDecl`s into a single iterable that then has the locals generated for it.

**Pull request merged**